### PR TITLE
[cond] make cond call fake kernel in dynamo

### PR DESCRIPTION
**Pull request by ydwu4**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #147130
* __->__ #147045
* #146954

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov @StrongerXi

### CODE REVIEW

File: test/functorch/test_control_flow.py [hunk 1]
```diff
@@ -578,22 +576,20 @@ def f(pred, x):
             grad_out = torch.ones_like(result)
             return torch.autograd.grad(result, (x,), grad_out)
 
-        gm = make_fx(f, tracing_mode="symbolic")(pred, x)
```
Comment (ydwu4) [inline test/functorch/test_control_flow.py:581]: The dispatching logic looks like this: in dynamo cond_op sees a functional_tensor[FakeTensor], then wrap_fx_proxy calls the cond_op tries to get its fake output, this triggers 1. autograd key (since some inputs require gradient), 2. functionalization key, then 3. fake_tensor_mode. For 1, we'll reenter_make_fx  fw and backward graph, which creates a new fake tesnor mode that's different from the fake tesnor mode on tensors (which is the dynamo fake tensor mode) and causing a mix fake tensor mode error.

Since the fix will make the tracing logic of autograd key even more hacky and we'll refactor the autorad key so that it doesn't materialize the graph soon, we just turn off symbolic tracing mode since it's not used anyway.

File: torch/_dynamo/variables/higher_order_ops.py [hunk 2]
```diff
@@ -959,18 +959,12 @@ def speculate_branch(branch):
...
-            true_r.as_proxy(),
-        )
-
         return _call_function_and_unflatten_output(
             tx,
             torch.ops.higher_order.cond,
             p_args,
             {},
-            flat_example_value,
+            None,
```
Comment (ydwu4) [inline torch/_dynamo/variables/higher_order_ops.py:967]: By not providing an example value, wrap_fx_proxy will call the hop with the fake inputs, which re-uses all the dispatch key implementations so that we don't need to have redundant code that merge the fake outputs in different places.

Comment (zou3519) [reply torch/_dynamo/variables/higher_order_ops.py:986]: nice

**Review by zou3519 — APPROVED** [review]
